### ISSUE TITLE
Automatic Departure Reminders to file Handlers and their Managers

### ISSUE THREAD

**Issue by chris206** [206Tours/DB#1781]
At 100 days prior, an email to be automatically generated for the file handler and their manager for review.  At 70 days prior, emails will be generated to file handler only UNLESS the departure is not at least +/- 20% of expected number.
At 40 days prior, Ticketing/Documents reminder will be generated for handlers & managers.

To implement this correctly, we will need a way to associate a manager to each user, which can be input in their user account time of hiring & modified during the course of employment.

SEE: 
Create Manager Association in Administrative Panel #1764
Calculate Days Prior to Departure in "All Departures" #1780

**Comment by chris206** [comment]
Can you create an administrative notification section separate from the client notification?

I want to eliminate any crossover or confusion

**Comment by bradatkins** [comment]
There was no mention of a client notification? it was handlers and their managers?

**Comment by chris206** [comment]
correct - but I was guessing that you would put the messages in the general messages center. I just wanted to make sure that internal email messages are separate from existing client messages

**Comment by bradatkins** [comment]
Can you rephrase this to be clearer?

At 70 days prior, emails will be generated to file handler only UNLESS the departure is not at least +/- 20% of expected number of pilgrims

**Comment by chris206** [comment]
scratch this idea - not necessary.


At 100 days prior, an email to be automatically generated for the file handler and their manager for review. 
Title: 95 Days Prior - File Review: GROUP NAME | DESTINATION | DATE
Body: Group Air & Materialization, (include data from land and air in Departure code.)

At 70 days prior, emails will be generated just to file handler
Title: 60 Days Prior - GROUP NAME | DESTINATION | DATE 
Body: Accounts Past Due, Build KBYG, Ensure Air Reservations

At 40 days prior, Ticketing/Documents reminder will be generated for handlers & managers.
Title: 30 Days Prior - GROUP NAME | DESTINATION | DATE 
Body: Issue airfare, send documents

**Comment by bradatkins** [comment]
OK.

**Comment by chris206** [comment]
I changed the main request to reflect some of the updated info.

there is an important exclusion

**Comment by bradatkins** [comment]
What is a sub code?

**Comment by chris206** [comment]
#1803 The TPP code that you are working on.

These are weird hybrid tours that may have group space because they are all traveling together, but will be merging with series tours

**Comment by bradatkins** [comment]
OK. I haven't heard back from Sandy / Milanka on approval to proceed yet, so I'm just waiting on that. If you happen to go by, feel free to mention it :)

**Comment by chris206** [comment]
mentioned it, did you get an email from Sandy?

**Comment by bradatkins** [comment]
Yup, made some adjustments just waiting to hear back.

**Comment by bradatkins** [comment]
This is done; however, I have no enabled the job just yet. You need to

1) Finish associating employees to Managers
2) Modify the messages for 100, 70, 40 day review in the admin
3) Ensure tour departures have the # Expected inputted

**Comment by chris206** [comment]
OK - here is the tag map. Please import that info to the message and activate each.
![image](https://cloud.githubusercontent.com/assets/22382144/23468198/2b2a583e-fe6e-11e6-82e4-45c0feb3bf76.png)

**Please exclude SERIES departures from these notifications, otherwise they will get 50 messages a day**

Over time, we will modify wording as necessary

**Comment by chris206** [comment]
The # expected shouldn't be required for the message to go out to file manager, just overall manager, right?

also - dont send them for cancelled departures

**Comment by bradatkins** [comment]
Correct, it's not required

**Comment by chris206** [comment]
OK! Please just add tags ass follows:

![image](https://cloud.githubusercontent.com/assets/22382144/24766204/7289126c-1ac8-11e7-9c45-cec0f038f176.png)

**Comment by chris206** [comment]
Hey Brad,

How are we doing with this one? is it nearing implementation?

![image](https://cloud.githubusercontent.com/assets/22382144/24963545/e4e3ed02-1f6c-11e7-8c6b-4923835e09c5.png)

**Comment by bradatkins** [comment]
It's ready, just disabled until you give me the thumbs up :)

**Comment by chris206** [comment]
If the tags are added as per the prior...
https://img.clipartfest.com/c58a72d7bb89c340293d782585758f58_24da0f778f40de98d080f45e1b57b8-thumbs-up-clipart-free_1307-1600.png
https://img.clipartfest.com/c58a72d7bb89c340293d782585758f58_24da0f778f40de98d080f45e1b57b8-thumbs-up-clipart-free_1307-1600.png

**Comment by chris206** [comment]
So, will Amanda and Bika be receiving their 40 day file reminder for SD052617 this Sunday?

**Comment by chris206** [comment]
![image](https://cloud.githubusercontent.com/assets/22382144/25099142/191d19d0-2379-11e7-9768-54ef2a9aed0f.png)

**Comment by chris206** [comment]
Only include SERIES departures in this reminder if there are any numerical values in the Air Arrangments

**Comment by bradatkins** [comment]
Sorry, ya I saw that screenshot, I'm not sure what you mean by "pull from Air in Departure Code" and Ground as well, can you take a screenshot of the actual fields in a departure you are talking about?

**Comment by chris206** [comment]
From TOURS:
1. Tour #
2. Tour Name

From TOURDEPARTURES:
3. Start Date
4. End Date
5. Ground Operator
6. Air Operator

![image](https://cloud.githubusercontent.com/assets/22382144/25100178/1dc645c0-237d-11e7-81df-87dde6d0e515.png)

![image](https://cloud.githubusercontent.com/assets/22382144/25100194/283e12ee-237d-11e7-8309-c125ee37da88.png)

**Comment by bradatkins** [comment]
The job is now enabled.

**Comment by chris206** [comment]
this is great. 

We dont need the closing info; sincerely, contact info etc
Can you separate the subject line with | ?- so:

40 Day Review | Tour # | Aug 7 - 12, 2017

**Comment by bradatkins** [comment]
Done

**Comment by chris206** [comment]
It is currently sending reminders for departures that are cancelled in the database.

instead of NOT doing this, can we tag the Status of the Departure in the top of the email:

"This Departure is ACTIVE/CANCELLED/SOLD OUT"

**Comment by chris206** [comment]
Also, "www.206tours.com" is still populating right on top of the air arrangements at the bottom of the email

**Comment by chris206** [comment]
Also, the subject line reads "206 Tours Inc 40 Day Review" this is not necessary.
Is this a rule for all of our emails?
